project-management/Github.md

```diff
@@ -0,0 +1,77 @@
+# Managing public (or private) projects using GitHub
+
+We use a common set of labels across projects in GitHub. It is
+possible to precisely control this set of labels using
+the [settings GitHub app][github-app-settings] (needs to be enabled
+for your org). Use the [`settings.yml`](../.github/settings.yml) from
+this repository as a starting point.
+
+[github-app-settings]: https://github.com/probot/settings
+
+## Labels
+
+We use labels to classify tasks by priority, from P0 to P4:
+
+* **P0 (blocker)** Do not do anything else, this is a blocker.
+* **P1 (critical)** Must have.
+* **P2 (major)** Should have.
+* **P3 (minor)** Could have.
+* **P4 (trivial)** Someday/maybe. Consider closing as "wontfix".
+
+Labels are also useful to classify issues by type or by component. Use
+colours to distinguish whether a label designates a type, a component
+or something else.
+
+## Project leads
+
+Each project must have a project lead (maintainer). By convention, the
+PL is whoever appears at the top of [`CODEOWNERS`][github-codeowners]
+file, e.g.:
+
+```
+* @mboes
+```
+
+This has the effect that @mboes will automatically be tagged as
+a reviewer for all incoming PR's, unless later lines in the same file
+override that.
```

Comment:
Should we make a standard policy to lock merge before every code owners have acknowledge?

I'm wondering if using the codeowner thing is not a bit too bureaucratic anyway. It tends to over assign, and may yield more « you were assigned as reviewer because you are a code owner » notification than is really useful.

On the other hand, I suppose that it is super useful for third-party contributors (of open source projects) which may not know whom to assign as reviewer.

Reply:
Point is, all projects have maintainers. My experience is that maintainers themselves forget over time which open source projects they're maintainers for. I have to keep reminding people myself and that's not something I want to be doing in the future. It's useful to put down *in writing* somewhere who that maintainer is. The `CODEOWNERS` file is an existing mechanism to do just that (though a bit more general). We're just asking that the file be non-null. This is a must for open source projects. For private projects, as with everything, use your judgment. But if you're the PL on a project then you should be aware of every change that goes in.

> Should we make a standard policy to lock merge before every code owners have acknowledge?

Worth considering later. Need more data first.

Reply:
> Worth considering later. Need more data first.

It's worth saying then: « We don't have a company policy regarding whether the <codeowner locking option> should be turned on. Feedback is welcome. ».